init.php

```diff
@@ -9,28 +9,26 @@
 define('DEBUG_MODE', TRUE);
 
 //limit the number of forked processes to the number of available cpu cores
-function cpu_cores() {
-  $cores = 1;
-  if (is_file('/proc/cpuinfo')) {
-    $cpuinfo = file_get_contents('/proc/cpuinfo');
-    preg_match_all('/^processor/m', $cpuinfo, $matches);
-    $cores = count($matches[0]);
-  } else if ('WIN' == strtoupper(substr(PHP_OS, 0, 3))) {
-    $process = popen('wmic cpu get NumberOfLogicalProcessors', 'rb');
-    if (false !== $process) {
-      fgets($process);
-      $cores = intval(fgets($process));
-      pclose($process);
-    }
-  } else {
-    $process = popen('sysctl -a', 'rb');
-    if (false !== $process) {
-      $output = stream_get_contents($process);
-      preg_match('/hw.ncpu: (\d+)/', $output, $matches);
+function cpu_cores($cores = 1) {
```

Comment:
I don't think we need `$cores` as a function parameter.

Reply:
Yes, I put it there to lower the LoC count and initialize it with a default value at the same time. Anyway it was a tough battle to run PHP and test the script on 64bit Windows XP. I was surprised to see that PHP (5.4.45 TS 32bit) runs it almost instantaneously while PHP 7.0.9 needed almost 2 seconds for that :\

Reply:
Honestly, I am impressed that you even bothered trying this on Windows :smile:

Reply:
Ye, Chrome died two times due to low memory ... those virtual machines are hungry